src/ray/object_manager/plasma/store.cc

```diff
@@ -230,10 +230,19 @@ uint8_t *PlasmaStore::AllocateMemory(size_t size, MEMFD_TYPE *fd, int64_t *map_s
     std::vector<ObjectID> objects_to_evict;
     int64_t space_needed = eviction_policy_.RequireSpace(size, &objects_to_evict);
     EvictObjects(objects_to_evict);
-    // More space is still needed. Try to spill objects to external storage to
-    // make room.
+    // More space is still needed.
     if (space_needed > 0) {
-      *error = PlasmaError::OutOfMemory;
+      RAY_LOG(DEBUG) << "attempt to allocate " << size << " failed, need " << space_needed
+                     << " num bytes unsealed: " << num_bytes_unsealed_;
+      space_needed -= num_bytes_unsealed_;
+      if (space_needed > 0) {
+        // Even after all unsealed objects are sealed, there would not be
```

Comment:
Can you also comment why sealing unsealed objects will make space? It is because they are spillable after that right?